File: Richie/Valid Parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        open_ch = ["(", "{", "["]
        closed_ch = [")", "}", "]"]
        match = {
            ")": "(",
            "}": "{",
            "]": "["
        }
        
        for ch in s:
            if ch in closed_ch and len(stack) == 0:
                return False
            if ch in closed_ch and match[ch] == stack[-1]:
                stack.pop()
                continue
                
            stack.append(ch)
        
        return len(stack) == 0
        
#

File: Engr-T.stark/validParentheses.cpp
bool isValid(string s) {
        stack<char> st;
        for(char& c:s)
        {
            if(c=='(' || c=='{' || c=='['){st.push(c);}   
            else {
                if(st.empty()) return false;
                else if(c==')' && st.top()!='(' )return false;
                else if(c=='}' && st.top()!='{' )return false;
                else if(c==']' && st.top()!='[' )return false;
                st.pop();
            }
        }
        return st.empty();
    }
